automation/src/tests/navigation/navigation-menu.spec.ts

```diff
@@ -0,0 +1,66 @@
+import PuppeteerLaunch, {userAgent} from '../../fixtures/puppeteer-launch';
+import HomePage from '../../pages/HomePage';
+import LoginPage from '../../pages/LoginPage';
+import ProfilePage from '../../pages/ProfilePage';
+import WorkspacesPage from '../../pages/WorkspacesPage';
+
+jest.setTimeout(60 * 1000);
+
+describe('Page navigation-menu tests:', () => {
+
+  let browser;
+  let incognitoContext;
+  let page;
+
+  beforeAll(async () =>  {
+    browser = await PuppeteerLaunch();
+  });
+
+  beforeEach(async () => {
+    incognitoContext = await browser.createIncognitoBrowserContext();
+    page = await incognitoContext.newPage();
+    await page.setUserAgent(userAgent);
+    await page.setDefaultNavigationTimeout(60000);
+    const loginPage = new LoginPage(page);
+    await loginPage.login();
+  });
+
+  afterEach(async () => {
+    await incognitoContext.close();
+  });
+
+  afterAll(async () => {
+    await browser.close();
+  });
+
+  test('User sign out', async () => {
+    const homePage = new HomePage(page);
+    await homePage.waitForReady();
+
+    await homePage.navigation.signOut();
+    expect(await page.url()).toContain('login');
+  });
+
+  test('User open Profile page', async () => {
+    const homePage = new HomePage(page);
+    await homePage.waitForReady();
+
+    await homePage.navigation.toProfile();
+
+    const profile = new ProfilePage(page);
+    await profile.waitForReady();
+    expect(await page.url()).toContain('profile');
+  });
+
+  test('User open all-workspaces page', async () => {
+    const homePage = new HomePage(page);
+    await homePage.waitForReady();
+
+    await homePage.navigation.toAllWorkspaces();
+
+    const workspaces = new WorkspacesPage(page);
+    await workspaces.waitForReady();
+    expect(await page.url()).toContain('workspaces');
```

Comment:
Small, single-interaction tests like this tend not to provide much value by themselves. Are we expecting that the 'See all workspaces' link is at such high risk of breaking that it needs an isolated test case?

A better use of this kind of automation is to string together longer flows that mimic how a user would actually interact with the site. Those kinds of tests can uncover bugs and edge cases that live within the complexity of the system as a whole, rather than simple coding mistakes that are easier to catch by other means.